olive/common/utils.py

```diff
@@ -162,3 +162,28 @@ def tensor_data_to_device(data, device: str):
         return set(tensor_data_to_device(v, device) for v in data)
     else:
         return data
+
+
+def get_attr(module, attr, fail_on_not_found=False):
+    """
+    Get attribute from module.
+
+    :param module: module to get attribute from
+    :param attr: attribute name, can be a string with dot notation. If empty, return modulenot
+    :param fail_on_not_found: if True, raise AttributeError if attribute is not found
+    :return: attribute
+    """
+    if not attr:
+        # return module if attr is empty
+        return module
+
+    attr = attr.split(".")
+    for a in attr:
+        try:
+            module = getattr(module, a)
+        except AttributeError:
+            if fail_on_not_found:
+                raise AttributeError(f"Attribute {attr} not found.")
```

Comment:
I suppose raise is enough. If we want to re-raise the exception, we could use raise from e

Reply:
The error message is different from the original one. The original message only mentions `a` but we want to raise with `attr` in the message.

Reply:
how about [https://stackoverflow.com/questions/24752395/python-raise-from-usage](https://stackoverflow.com/questions/24752395/python-raise-from-usage)?

Reply:
[https://github.com/microsoft/OLive/blob/9028fd5dd6def368ce93dda14140fe01fff8b3ed/olive/passes/onnx/quantization.py#L412](https://github.com/microsoft/OLive/blob/9028fd5dd6def368ce93dda14140fe01fff8b3ed/olive/passes/onnx/quantization.py#L412)

Reply:
Thanks! just updated it. 
The `raise from` method is much better since it returns a better trace.